### PR TITLE
Simplify the hash aggregation spill processing

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Simplify the hash aggregation spill processing by removing the
fine-grained partition control logic in grouping set:
- Always create aggregation spiller with single partition
- Always spill all the rows from spiller no matter it is triggered by memory arbitration,
   threshold or test injection
- Simplify the corresponding un-spilling code path.

The final or single aggregation is multi-threaded executed on a worker which means
that each aggregate operator runs a subset of data so that fine-grained partitioning
inside an operator is not that necessary.

The followup is to remove the corresponding query configs and simplification inside
the spiller.

### CODE REVIEW

**Comment by mbasmanova** [comment]
@xiaoxmeng This is a nice change. Some folks may wonder whether this implementation creates a steep performance cliff when query runs out of memory by a tiny bit. I believe this is not the case because aggregation runs multi-threaded and in case we are short of memory just a bit only on of the threads will be spilling, not all of them. Assuming this is the case, it would be nice to clarify this in the PR description.

File: velox/common/memory/MemoryArbitrator.h [hunk 1]
```diff
@@ -89,14 +89,6 @@ class MemoryArbitrator {
   /// registered.
   static void unregisterFactory(const std::string& kind);
 
-#ifdef VELOX_ENABLE_BACKWARD_COMPATIBILITY
```
Comment (mbasmanova) [inline velox/common/memory/MemoryArbitrator.h:92]: unrelated change?

File: velox/exec/GroupingSet.cpp [hunk 2]
```diff
@@ -848,23 +847,13 @@ void GroupingSet::ensureInputFits(const RowVectorPtr& input) {
...
   if (spillMemoryThreshold_ != 0 && currentUsage > spillMemoryThreshold_) {
-    const int64_t bytesToSpill =
-        currentUsage * spillConfig_->spillableReservationGrowthPct / 100;
-    auto rowsToSpill = std::max<int64_t>(
-        1, bytesToSpill / (rows->fixedRowSize() + outOfLineBytesPerRow));
-    spill(
-        std::max<int64_t>(0, numDistinct - rowsToSpill),
-        std::max<int64_t>(
-            0, outOfLineBytes - (rowsToSpill * outOfLineBytesPerRow)));
+    spill(0, 0);
```
Comment (mbasmanova) [inline velox/exec/GroupingSet.cpp:856]: It would be nice to add spill() API without any parameters.

**Review by mbasmanova — APPROVED** [review]
Looks good though I didn't read every line.

File: velox/exec/GroupingSet.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply velox/exec/GroupingSet.cpp:856]: Yes, I will add this in followup and deprecate the existing ones.

File: velox/common/memory/MemoryArbitrator.h (hunk 1 above)
Comment (xiaoxmeng) [reply velox/common/memory/MemoryArbitrator.h:92]: Reverted.
